### PR TITLE
add PostrgeSQL as pg, schemas, refactor

### PR DESCRIPTION
**Pull request by zhibirc**

### CODE REVIEW

File: src/db/index.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            const start = Date.now();
+            const result = await this.pool.query(sqlText, params);
+
+            this.logger.info(`Request "${sqlText}" completed in ${Date.now() - start}ms.`);
+
+            return result;
+        } catch ( exception ) {
+            this.logger.error(exception);
+
+            return null;
```
Comment (viktornord) [inline src/db/index.ts]: why we don't fail?

File: src/storage.ts [hunk 2]
```diff
@@ -1,146 +1,182 @@
...
+            );
 
-    update ( id: string, data: any ) {
-        const entry = this.__store.movies.find(item => item.imdbID === id);
+            return queryResult.rows[0].coalesce;
+        }
 
-        if ( entry ) {
-            const { comment, personalScore } = data;
+        queryResult = await this.db.query('SELECT movies.data FROM movies FULL JOIN meta ON movies.movie_id = meta.movie_id', []);
```
Comment (viktornord) [inline src/storage.ts]: why do you need full join if you only take data from movies table

**Review by viktornord — APPROVED** [review]
see comments

File: src/db/index.ts (hunk 1 above)
Comment (zhibirc) [reply src/db/index.ts]: Missed it, thanks. Fixed.

File: src/storage.ts (hunk 2 above)
Comment (zhibirc) [reply src/storage.ts]: Absolutely, I need all movie records, but some of movies can had comments (so called metadata) from another table. So, left join is appropriate here. Thanks! Fixed.

**Pull request merged**